File: CPP/bubble_sort.cpp
#include<iostream>
#include<cstdlib>
using namespace std;

void bubble_sort (int *d, int n){
   int i, j, tmp;
   cout << "number of elements:" << n << endl;   
   for (i=0; i < n; i++){
      for (j=i; j < n; j++){
         if (d[j] > d[i]) {
            tmp = d[i];   
            d[i] = d[j];
            d[j] = tmp; 
         }

      }
   }
}



int main (int argc, char *argv[]) {

    int *d; 
    int i;  
    d = new int [argc]; 
  
    for (i=1; i  < argc; i++) {
       d[i-1] = atoi(argv[i]); 
    }

    bubble_sort (d, argc-1);

    for (i=0; i < argc-1; i++)
        cout << d[i] << endl; 




}

File: Python/bubble_sort.py
import sys

def bubble_sort (d):
    for i in range (0, len(d)):
       for j in range(i, len(d)):
          if d[j] > d[i]:
             d[i], d[j] = d[j], d[i]
    return d 
 

if __name__ == "__main__":
 
    d = [int(sys.argv[i]) for i   in range (1, len(sys.argv))]

    print(bubble_sort(d))

